pkg/ebpf/tracee.go

```diff
@@ -1604,26 +1589,53 @@ func (t *Tracee) netEnabled() bool {
 // TODO: move to triggerEvents package
 //
 
-const uProbeMagicNumber uint64 = 20220829
-
 // triggerSyscallsIntegrityCheck is used by a Uprobe to trigger an eBPF program
 // that prints the syscall table
-func (t *Tracee) triggerSyscallsIntegrityCheck(event trace.Event) {
+func (t *Tracee) triggerSyscallsIntegrityCheck(event trace.Event) error {
 	_, ok := t.events[events.HookedSyscalls]
 	if !ok {
-		return
+		return nil
 	}
+
+	errArgFilter := make(map[int]error, 0)
+
+	for p := range t.config.Policies.Map() {
+		hookedSyscallsFilters := p.ArgFilter.GetEventFilters(events.HookedSyscalls)
+		if len(hookedSyscallsFilters) == 0 {
+			logger.Debugw("policy %d: no syscalls were provided to hooked_syscall event. "+
+				"using default configuration. please provide it via -f hooked_syscalls.args.check_syscalls=<syscall>,<syscall>", p.ID)
+			derive.SyscallsToCheck = events.DefaultSyscallsToCheck()
+		}
+
+		if len(derive.SyscallsToCheck) == 0 {
+			syscallFilter, ok := hookedSyscallsFilters["check_syscalls"].(*filters.StringFilter)
+			if syscallFilter != nil && ok {
+				eventNames := events.Definitions.NamesToIDs()
+				for _, syscall := range syscallFilter.Equal() {
+					_, ok := eventNames[syscall]
+					if !ok {
+						errArgFilter[p.ID] = fmt.Errorf("policy %d: %s - no such syscall", p.ID, syscall)
+						break
+					}
+					derive.SyscallsToCheck = append(derive.SyscallsToCheck, syscall)
+				}
+			}
+		}
+	}
+
+	for k, v := range errArgFilter {
+		if v != nil {
+			return errfmt.Errorf("error setting %v filter: %v", k, v)
+		}
+	}
+
 	eventHandle := t.triggerContexts.Store(event)
-	t.triggerSyscallsIntegrityCheckCall(
-		uProbeMagicNumber,
-		uint64(eventHandle),
-	)
+	t.triggerSyscallsIntegrityCheckCall(uint64(eventHandle), uint64(events.MaxSyscallID))
```

Comment:
So how can I know the size? Isn't it static for each architecture?

Reply:
BTW - we will read the memory after the syscall table but it will be relevant only if the user asks to check a syscall that he doesn't have in the system, and will get a wrong value. In short a really rare edge case.

Reply:
> BTW - we will read the memory after the syscall table but it will be relevant only if the user asks to check a syscall that he doesn't have in the system, and will get a wrong value. In short a really rare edge case.

And who says the memory is accessible and the read won't fail (probably not, but can you assure that?)
In a security product, you don't want to make assumptions and allow yourself not to think about edge cases.

I would use as a maximum the ID of the last syscall in kernel 4.18, which is the minimum kernel version we support. Then you can know that the table is, at least, that big. Of course, this also means that we won't support the newest system calls available on this system, which I think is good enough.